.github/workflows/tag-release.yaml

```diff
@@ -0,0 +1,46 @@
+name: Build and publish artifacts
+on:
+  push:
+    tags:
+      # Run on all tags (but not branches); releases artifact to Maven Central
+      # Note this is NOT REGEX - see https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet
+      - v[0-9]+.[0-9]+.[0-9]+
+jobs:
+  build-and-publish:
+    runs-on: ubuntu-22.04 # LTS EoL Apr 2025
+
+    env:
+      ORG_GRADLE_PROJECT_sonatypeUsername: ${{ secrets.SONATYPE_PUBLISH_USERNAME }}
+      ORG_GRADLE_PROJECT_sonatypePassword: ${{ secrets.SONATYPE_PUBLISH_PASSWORD }}
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          java-version: '8.x'
```

Comment:
8.x 👀

Reply:
Yeah, this is for maximum compatibility with downstream dependencies, some of which still support anything >= 8.